src/result.ts

```diff
@@ -23,14 +23,16 @@ export namespace Result {
     fn: Fn,
     errorFn?: (e: unknown) => E,
   ): (...args: Parameters<Fn>) => Result<ReturnType<Fn>, E> {
-    return (...args) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return ((...args: any[]) => {
       try {
         const result = fn(...args)
         return ok(result)
       } catch (e) {
         return err(errorFn ? errorFn(e) : e)
       }
-    }
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    }) as any
```

Comment:
Can you explain the rationale for these `as any` changes made here and in other files?

Reply:
Moving to strict: true revealed some errors that I didn't have time to debug. I suggest we 'as any' them now, and fix them in a followup PR.

Reply:
The reason this errors and you have to `as any` it is because with this call signature you can in theory do something like:

```ts
const safeParse = fromThrowable<typeof JSON.parse, number>(JSON.parse);
      // ^ (text: string, reviver?: (this: any, key: string, value: any) => any) => Result<any, number>
```

So by allowing providing a specific generic type as E without providing an errorFn, you can get in trouble. You can then end up in a case where you try to pass `e` to the `err` in the catch block (in the second/false ternary case), but since the user might have specified a specific E (like number in the example above), you can't just give it the unknown (or any here I suppose)`e`.

The reason why `strict: true` triggers this is because previously the `errorFn` didn't get the `| undefined` from the optional notation on the parameter `errorFn?`, since those does not affect the type without `strict` turned on. So basically typescript never considered that `errorFn` could be falsy, thus never hitting the `err(e)` code path of the ternary in the catch block, until strict was turned on.

The easiest fix might be to add some overloads  to prevent using the function as I did above:

```ts
export function fromThrowable<Fn extends (...args: readonly any[]) => any>(
  fn: Fn
): (...args: Parameters<Fn>) => Result<ReturnType<Fn>, unknown>;
export function fromThrowable<Fn extends (...args: readonly any[]) => any, E>(
    fn: Fn,
    errorFn: (e: unknown) => E,
  ): (...args: Parameters<Fn>) => Result<ReturnType<Fn>, E>;
  export function fromThrowable<Fn extends (...args: readonly any[]) => any, E>(
    fn: Fn,
    errorFn?: (e: unknown) => E,
  ): (...args: Parameters<Fn>) => Result<ReturnType<Fn>, E | unknown> {
    // implementation
}
```

I don't know if it really makes any practical difference here, but I believe it would be more correct to do `catch (e: unknown)` to have `e` typed as unknown and not the default `any`.

This could of course happen in a different PR. And I suppose in theory changing the call signature as I proposed might be considered a breaking change. However, if someone is using the function as I did above they should stop doing as it probably gives you types that don't match the runtime types. So you could consider it a bug maybe 🤷

Reply:
I created a separate PR for fixing the issue: https://github.com/supermacro/neverthrow/pull/583 However, feel free to close that PR and include the changes here instead if that's better.